Control.c

```diff
@@ -44,25 +56,59 @@ int main(void)
 	
 	while(1)
 	{
-		// Clear out the screen, and update
-		GLCD_ClearScreen();
-		LCD_UpdateScreenStatus();
-		LCD_UpdateScreenInsulin();
-		
-		// Wait for a short period of time before updating
-		for(i = 0; i < 150000; i++)
+		switch(Control_GlobalState)
 		{
-			for(j = 0; j < 25; j++);
+			case Undefined:
+			case Administration:
+				// Clear out the screen, and update
+			GLCD_ClearScreen();
+			LCD_UpdateScreenStatus();
+			LCD_UpdateScreenState();
+			LCD_UpdateScreenInsulin();
+			
+			// Wait for a short period of time before updating
+			for(i = 0; i < 150000; i++)
+			{
+				for(j = 0; j < 25; j++);
+			}
+				break;
+			case Empty:
+				do {
+					empty = Joystick_GetState();
+				} while(empty != 0x00800000);
```

Comment:
You forgot to mask the important bit. There is no logic here to even check if the Joystick is in a certain state, because you are comparing all the bits, not the 1 important bit.